dmoj/judgeenv.py

```diff
@@ -309,6 +296,10 @@ def get_supported_problems_and_mtimes(warnings=True):
                 else:
                     problem_dirs[problem] = problem_dir
                     problems.append((problem, os.path.getmtime(problem_dir)))
+
+    _problem_roots_cache = root_dirs
```

Comment:
Can you remove the `root_dirs` and replace it with `_problem_roots_cache = list(root_dirs_set)`

Reply:
We need to maintain the order of priority

Reply:
oh yeah totally forgot about that, then could you help comment it in the code? 🤔